File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/PaymentMapper.java
package com.td.wallendar.utils.mappers;

import com.td.wallendar.dtos.response.PaymentResponse;
import com.td.wallendar.models.Payment;

import java.util.HashSet;
import java.util.Set;

public class PaymentMapper {
    public static Payment toModel(PaymentResponse paymentResponse) {
        return new Payment(
                paymentResponse.getId(),
                ApplicationUserMapper.toModel(paymentResponse.getFrom()),
                ApplicationUserMapper.toModel(paymentResponse.getTo()),
                paymentResponse.getAmount(),
                paymentResponse.getDate()
        );
    }

    public static Set<Payment> toModel(Set<PaymentResponse> paymentResponseList) {
        Set<Payment> payments = new HashSet<>();
        for (PaymentResponse paymentResponse : paymentResponseList) {
            payments.add(toModel(paymentResponse));
        }
        return payments;
    }
}

File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/PaymentMapper.kt
package com.td.wallendar.utils.mappers

import com.td.wallendar.dtos.response.PaymentResponse
import com.td.wallendar.models.Payment
import java.util.*

object PaymentMapper {
    fun toModel(paymentResponse: PaymentResponse): Payment{
        return Payment(
                paymentResponse.id,
                ApplicationUserMapper.toModel(paymentResponse.from),
                ApplicationUserMapper.toModel(paymentResponse.to),
                paymentResponse.amount,
                paymentResponse.date
        )
    }

    fun toModel(paymentResponseList: MutableSet<PaymentResponse>): MutableSet<Payment>{
        val payments: MutableSet<Payment> = HashSet()
        for (paymentResponse in paymentResponseList) {
            payments.add(toModel(paymentResponse))
        }
        return payments
    }
}